08.ls_object/ls_file.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'etc'
+
+class LsFile
+  DIRECTORY_NUMBER = '40'
+
+  PERMISSION_MAP = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  TYPE_MAP = {
+    '01' => 'p',
+    '02' => 'c',
+    '40' => 'd',
+    '06' => 'b',
+    '10' => '-',
+    '12' => 'l',
+    '14' => 's'
+  }.freeze
+
+  attr_reader :name, :file_stat
+
+  def initialize(file_name)
+    @name = file_name
+    @file_stat = File.stat(file_name)
+  end
+
+  def type
+    TYPE_MAP[@file_stat.mode.to_s(8).slice(0..1)]
```

Comment:
```suggestion
    TYPE_MAP[@file_stat.mode.to_s(8).first(2)]
```

とした方がちょっとリーダブルかも（好みのレベルです）

Reply:
文字列クラスのfirstメソッドを増やすということでしょうか...？
こちらは単純な変更ではエラーが出てしまったので、一旦そのままにしました！

Reply:
あ、すいません、Railsの拡張と勘違いしてたみたいです 🙏 このままでOKです！
https://api.rubyonrails.org/classes/String.html#method-i-first